### DISCUSSION TITLE
UPDI as digital input

### DISCUSSION THREAD

**Discussion by Yardie-** [SpenceKonde/megaTinyCore#380] [Q&A]
Just wondering if it is possible to use the UPDI pin as a digital input on the 3216. 
ie after a certain period (leaving a programming window after boot) set the pin as input then just do a digital read in the loop
Would setting the pin as input disrupt the UPDI functionality.
This is of course maintaining the normal functionality.
I thought I saw something on this in the old docs but I can't find it now.

**Comment by grandaspanna** [comment]
I have used it as an input successfully. As long as your input doesn't look anything like the programming sequence, it should work fine.

**Reply by Yardie-** [reply]
Thanks. 
I just wanted to add a simple push button process trigger which would be pulled low with a weak resistor. The switch would pull the pin high. The whole thing would be removed to program the chip anyway.

**Comment by SpenceKonde** [comment]
Huh! Cool! I... I did not realize that worked...... I thought I tried it and determined that it always read high. But it is definitely working for me now. 

Don't do it that way, have the pin pulled up normally and the switch ground it. Always have buttons and switches pull down (there's even a builtin pullup on the UPDI pin, though it's a little weaker than you probably want - though if you don't want to interfere with UPDI programming it's gotta be pretty weak, those pin drivers are terribly weak....

**Reply by Yardie-** [reply]
ok 
Thanks Spence that's pretty simple. 
Wait for a bit after boot ( tobe sure tobe sure tobe sure) then make the pin INPUT_PULLUP and watch for low in the loop.
I wonder can you attach an itterrupt to it :D 
But polling will be fine for most jobs anyway.

Nice simple little use of the pin and the breakout pins as well (which is what I do)

Thanks @grandapanna  and Spence

**Comment by grandaspanna** [comment]
I made the mistake of putting a p-channel mosfet in front of the UPDI pin for a bit of decoupling and then wondered why I couldn't program it. Who knew mosfets were one way devices? My bad :-)
